### ISSUE TITLE
Duplicate uploads in single run of build_release.sh script

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/policybrain-builder#48]
I just used the `build_release.sh` script successfully to create and upload to anaconda.org/ospc fifteen taxcalc packages for Tax-Calculator release 0.9.1.  So, the script does work for me in building and uploading taxcalc packages.  Thank you, @PeterDSteinberg, for creating this capability.

As I get more experience with `build_release.sh`, I have some thoughts for improvement.  This issue describes an efficiency issue that will make the script run more smoothly and without any warnings.  I will raise another issue about the basic philosophy of the `build_release.sh` script in another issue.

The efficiency issue I want to report here is probably the same issue noted in #42, but my experience suggests a different cause of the warnings (they are not "errors", which is the terminology used in issue #42).  Basically, the logical structure of the `build_release.sh` script is now such that when building and uploading packages for the five operating systems and the three versions of Python we are supporting (2.7, 3.5, and 3.6), the five packages for 2.7 get uploaded three times, the five packages for 3.5 get uploaded two times, and the five packages for 3.6 get uploaded once.  The unnecessary duplicate uploads trigger a WARNING that doesn't generate confidence in new users of the script.

Also, it seems as if all five OS versions of a package are generated even if the script is skipping the uploads, which is another inefficiency.

Surely the script can be revised to do build the four other OS packages only if uploading and uploading each of the fifteen packages only once.

The `build_release.sh` script generates quite a bit of output, which is good, but can obscure what is going on.  When I built and uploaded the fifteen packages for taxcalc 0.9.1, I filtered the script output to make the duplicate uploads stand out.  Here is what I got this morning:
```
iMac2:policybrain-builder mrh$ . ./build_release.sh 2>&1 | awk '$1~/Upload/||$1~/WARNING/||$1~/OK/'
Uploading file ospc/taxcalc/0.9.1/osx-64/taxcalc-0.9.1-py27_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/linux-32/taxcalc-0.9.1-py27_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/linux-64/taxcalc-0.9.1-py27_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/win-32/taxcalc-0.9.1-py27_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/win-64/taxcalc-0.9.1-py27_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/osx-64/taxcalc-0.9.1-py35_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/linux-32/taxcalc-0.9.1-py27_0.tar.bz2 ... 
[WARNING] Distribution linux-32/taxcalc-0.9.1-py27_0.tar.bz2 already exists ... removing
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/linux-32/taxcalc-0.9.1-py35_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/linux-64/taxcalc-0.9.1-py27_0.tar.bz2 ... 
[WARNING] Distribution linux-64/taxcalc-0.9.1-py27_0.tar.bz2 already exists ... removing
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/linux-64/taxcalc-0.9.1-py35_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/win-32/taxcalc-0.9.1-py27_0.tar.bz2 ... 
[WARNING] Distribution win-32/taxcalc-0.9.1-py27_0.tar.bz2 already exists ... removing
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/win-32/taxcalc-0.9.1-py35_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/win-64/taxcalc-0.9.1-py27_0.tar.bz2 ... 
[WARNING] Distribution win-64/taxcalc-0.9.1-py27_0.tar.bz2 already exists ... removing
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/win-64/taxcalc-0.9.1-py35_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/osx-64/taxcalc-0.9.1-py36_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/linux-32/taxcalc-0.9.1-py27_0.tar.bz2 ... 
[WARNING] Distribution linux-32/taxcalc-0.9.1-py27_0.tar.bz2 already exists ... removing
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/linux-32/taxcalc-0.9.1-py35_0.tar.bz2 ... 
[WARNING] Distribution linux-32/taxcalc-0.9.1-py35_0.tar.bz2 already exists ... removing
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/linux-32/taxcalc-0.9.1-py36_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/linux-64/taxcalc-0.9.1-py27_0.tar.bz2 ... 
[WARNING] Distribution linux-64/taxcalc-0.9.1-py27_0.tar.bz2 already exists ... removing
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/linux-64/taxcalc-0.9.1-py35_0.tar.bz2 ... 
[WARNING] Distribution linux-64/taxcalc-0.9.1-py35_0.tar.bz2 already exists ... removing
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/linux-64/taxcalc-0.9.1-py36_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/win-32/taxcalc-0.9.1-py27_0.tar.bz2 ... 
[WARNING] Distribution win-32/taxcalc-0.9.1-py27_0.tar.bz2 already exists ... removing
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/win-32/taxcalc-0.9.1-py35_0.tar.bz2 ... 
[WARNING] Distribution win-32/taxcalc-0.9.1-py35_0.tar.bz2 already exists ... removing
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/win-32/taxcalc-0.9.1-py36_0.tar.bz2 ... 
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/win-64/taxcalc-0.9.1-py27_0.tar.bz2 ... 
[WARNING] Distribution win-64/taxcalc-0.9.1-py27_0.tar.bz2 already exists ... removing
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/win-64/taxcalc-0.9.1-py35_0.tar.bz2 ... 
[WARNING] Distribution win-64/taxcalc-0.9.1-py35_0.tar.bz2 already exists ... removing
Upload(s) Complete
Uploading file ospc/taxcalc/0.9.1/win-64/taxcalc-0.9.1-py36_0.tar.bz2 ... 
Upload(s) Complete
OK
iMac2:policybrain-builder mrh$ 
```

@MattHJensen @PeterDSteinberg @jbcrail

**Comment by martinholmer** [comment]
Pull request #52 by @jbcrail resolved issue #48.  Thanks, @jbcrail.
